core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java

```diff
@@ -1173,6 +1173,13 @@ public void prepare(MethodContext context) {
                 nonDefaultConstructorHandles[i] = loadObjectInstance(obj, existing,
                         parameterTypes[count++], relaxedValidation);
             }
+            if (nonDefaultConstructorHolder.constructor.getParameterCount() > 0) {
+                Parameter[] parameters = nonDefaultConstructorHolder.constructor.getParameters();
+                for (int i = 0; i < parameters.length; ++i) {
+                    String name = parameters[i].getName();
```

Comment:
I think we need to check if `java.lang.reflect.Parameter#isNamePresent` here, WDYT @Ladicek ?

Reply:
I was wondering why the existing code (which I merely copied) doesn't check if parameter names exist, but it seemed to work... :shrug:

Yeah, I agree we shouldn't put the synthetic parameter names `argN` to the map. That, together with how the map is used (matching parameters are removed from it, and then the map is checked if some parameters remain), should fix the issue here.

Reply:
@Ladicek mind pushing the fix to this PR?

Reply:
Pushed a commit, let's see how that goes :-)